src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs

```diff
@@ -84,32 +225,40 @@ public static void UseCastleWindsor(this IApplicationBuilder app, IWindsorContai
 			{
 				var resolve = container.Resolve<T>();
 				await resolve.InvokeAsync(context, async (ctx) => await next());
+				container.Release(resolve);
```

Comment:
It is probably worth a try/finally in case InvokeAsync throws.

Reply:
Yep. Good shout. Done.